core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -2067,6 +2070,10 @@ else if (sourceNode instanceof SampleNode sampleNode) {
                 throw e;
             }
             catch (RuntimeException e) {
+                if (Throwables.getRootCause(e) instanceof MethodTooLargeException) {
+                    throw new TrinoException(QUERY_EXCEEDED_COMPILER_LIMIT,
+                            "Query exceeded maximum filters. Please reduce the number of filters referenced and re-run the query.", e);
```

Comment:
It might not be specific to filters right ? Projection expression could still cause this issue

Reply:
What are you suggesting?  The approach here was to take the cases where COMPILER_ERROR is the error code and to separate out `MethodTooLargeException` into it's own case.

Reply:
It should use the same error message as below, just with a different error code.

Reply:
And similar to the other places that handle compiler failures, the below shouldn't have a message, as that would likely be an actual bug in the compiler code. So we should move the message from below to here.